Rust/Savina/src/concurrency/BoundedBuffer.lf

```diff
@@ -18,74 +18,70 @@ import BenchmarkRunner from "../lib/BenchmarkRunner.lf";
 
 reactor ManagerReactor(bufferSize: usize(50), numProducers: usize(40), numConsumers: usize(40)) {
     state num_producers(numProducers);
-    
+
     state adjusted_buffer_size: usize({=bufferSize-numProducers=});
     state pending_data: VecDeque<f64>;
     state num_terminated_producers: usize(0);
     state producer_terminated: Vec<bool>;
     state producer_id: usize(0);
-    
+
     input start: unit;
     output finished: unit;
-    
+
     input[numProducers] producerData: f64;
     input[numProducers] producerFinished: unit;
     output[numProducers] producerCommand: unit;
-    
+
     input[numConsumers] consumerAvailable: unit;
     output[numConsumers] consumerData: f64;
-    
+
     preamble {=
         use std::collections::VecDeque;
     =}
-    
+
     reaction(start) -> producerCommand {=
         // reset local state
         self.pending_data.clear();
         self.num_terminated_producers = 0;
         self.producer_terminated = vec![false; self.num_producers];
         self.producer_id = 0;
-        
+
         for port in producerCommand {
             ctx.set(port, ());
         }
     =}
-    
+
     reaction(consumerAvailable) -> consumerData, producerCommand, finished {=
         // abort and signal finished if all producers have terminated and all data has been send
         if self.num_terminated_producers == self.num_producers && self.pending_data.is_empty() {
             ctx.set(finished, ());
         } else {
-            let mut consumer_id = 0;
             let mut activated_producers = 0;
-            
+
             for (consumer_available, consumer_data) in consumerAvailable.into_iter().zip(consumerData) {
                 if ctx.is_present(&consumer_available) {
                     if let Some(data) = self.pending_data.pop_front() {
                         ctx.set(consumer_data, data);
-                        
+
                         if activated_producers < self.num_producers && !self.producer_terminated[self.producer_id] {
                             ctx.set(producerCommand.get(self.producer_id), ());
                             self.producer_id = (self.producer_id + 1) % self.num_producers;
-                            activated_producers += 1;                        
+                            activated_producers += 1;
                         }
-                    }        
-                } else {
-                    break;
+                    }
                 }
-                consumer_id += 1; 
```

Comment:
This shouldn't be removed though, right?

When you merge this with master (or rebase) the tests should also pass again.

Reply:
This was a commit I forgot on the main repository for a few months and noticed today. But from what I can tell this is a local variable that isn't used anywhere. It's just an artefact from porting the benchmark and isn't needed any more.

Reply:
Oh, I see